### PR TITLE
Fix final reward to follow the spec

### LINKED ISSUE #1609: Reward curve does not match with spec.
reward_curve

I drew the two graphs: one following the spec (blue line), the other following the implementation(red line). They were not be completely folded up.  We should change one following the other.

### PR DESCRIPTION
**Pull request by HoOngEe**
It closes #1609

### CODE REVIEW

File: core/src/consensus/tendermint/engine.rs [hunk 1]
```diff
@@ -349,22 +349,26 @@ fn calculate_pending_rewards_of_the_previous_term(
 /// reward = floor(intermediate_rewards * (a * number_of_signatures / number_of_blocks_in_term + b) / 10)
 fn final_rewards(intermediate_reward: u64, number_of_signatures: u64, number_of_blocks_in_term: u64) -> u64 {
     let (a, b) = if number_of_signatures * 3 >= number_of_blocks_in_term * 2 {
-        // number_of_signatures / number_of_blocks_in_term >= 2 / 3
+        // number_of_signatrues / number_of_blocks_in_term >= 2 / 3
```
Comment (sgkim126) [inline core/src/consensus/tendermint/engine.rs]: typo

Comment (HoOngEe) [reply core/src/consensus/tendermint/engine.rs]: the typo fixed

**Comment by sgkim126** [comment]
@HoOngEe A unittest failed.

**Comment by HoOngEe** [comment]
@sgkim126 I fixed my mistake omitting minus sign and fixed test cases to follow the spec.

**Comment by sgkim126** [comment]
@HoOngEe Please check travis.

**Comment by HoOngEe** [comment]
@sgkim126 Sorry, I made a typo.

**Comment by sgkim126** [comment]
@HoOngEe Would you merge 5b17d100f61ee5c4f08dc5d2995996e85c167596 and 4f52634fb1f732b8e02553c256ad6671b3311793?

**Comment by HoOngEe** [comment]
@sgkim126 I merged those two commits into one and updated branch.

**Comment by sgkim126** [comment]
@HoOngEe What I meant was merge the commit to change the behavior and modify the test according to its behavior.

**Comment by HoOngEe** [comment]
@sgkim126 Two commits are merged as you explained.

**Pull request merged**